paywall fix

nynevi opened the pull request:
added an alert if the story has one section because paywalls cannot work with only one section. also lists cannot be broken into two parts due to technical limits

nynevi commented:
lists are split on backend so I was wrong, I will check the parsing algorithm to make it work with lists

nynevi commented:
it is almost impossible to correctly parse incorrect html so lists turn out to be wrong.

I matched the element count of frontend and backend so if user sees 1 section only, he cannot enable paywalls. there is a check for this on the backend if used via api

adrianbarwicki commented on src/editor/EditorCtrl.ts:
```diff
@@ -265,6 +266,9 @@ export default class EditorCtrl {
       fixedBody = this.editorService.getFixedBody(bodyEditor, externalHtml);
+      $scope.hasOnlyOneSection = $(fixedBody).length === 0 || $(fixedBody).length === 1 ?
+        true :
+        false;
```
Is it equivalent to:  `$(fixedBody).length === 0 || $(fixedBody).length === 1` ?

adrianbarwicki replied:
Is it the same as:  `$(fixedBody).length < 2` ?

adrianbarwicki commented on public/templates/layout_write.html:
```diff
@@ -81,9 +81,10 @@ <h4 class="modal-title" id="myModalLabel"><b>Prepare your story for readers</h4>
 								<label>
-									<input type="checkbox" ng-checked="draft.hasPaidSection" ng-click="togglePaidSection()"> <strong>I have a paid section in my story</strong>
+									<input type="checkbox" ng-checked="draft.hasPaidSection" ng-disabled="hasOnlyOneSection" ng-click="togglePaidSection()" ng-class="{'disabled': hasOnlyOneSection}"> <span ng-class="{'bold': !hasOnlyOneSection, 'disabled': hasOnlyOneSection}">I have a paid section in my story</span>
+									<strong ng-if="hasOnlyOneSection" style="color: red;">&nbsp;<i class="fa fa-exclamation-circle" aria-hidden="true"></i> Please split your story into two sections</strong>
```
What is the purpose of `&nbsp;`?

adrianbarwicki commented:
After a merge of another PR, there are merge conflicts.

nynevi commented:
solved conflicts

adrianbarwicki commented:
@nynevi could you name the next PRs by the convention <ticketnumber>-shortdesc.

adrianbarwicki commented:
@nynevi when the story cannot be spit into two and the paywall option is selected, the publish button should be disabled.

adrianbarwicki commented:
There are conflicts with dev after the linting merge. @nynevi

nynevi commented:
- fixed merge conflicts
- added bootstrap styling to currency inputs
- added a reset function for draft's paywall settings if the post later had its body changed to one section to automatically reset it

nynevi commented:
- solved merge conflicts

nynevi commented:
closing this as it is no longer necessary because the functionality is in v2 now
